drivers/platform/xilinx/i2c.c

```diff
@@ -205,11 +167,20 @@ int32_t i2c_init(struct i2c_desc **desc,
 	case IIC_PL:
 #ifdef XIIC_H
 		; /** Intentional empty statement */
-		struct pl_table_item *tab_check_pl;
-		tab_inst = pl_inst_tab_find(xinit->device_id);
-		if (tab_inst != -1) {
-			xdesc->instance = pl_table[tab_inst].instance;
-			pl_table[tab_inst].inst_no++;
+		struct inst_table_item tab_check_pl;
+		struct inst_table_item *temp_el_pl;
+
+		if (!pl_list)
+			list_init(&pl_list, LIST_DEFAULT, i2c_cmp);
+		if (!pl_it)
+			iterator_init(&pl_it, pl_list, true);
+
+		tab_check_pl.device_id = xdesc->device_id;
+		ret = iterator_find(pl_it, &tab_check_pl);
```

Comment:
```suggestion
		if (IS_ERR_VALUE(list_read_find(pl_list, (void **)&temp_el_pl, &tab_check_pl))
		      goto pl_error; //Item not found
                 else
                      //Item found and stored in temp_el_pl
```
You don't need to use an iterator. Same ps_it

Reply:
Will change.

Reply:
I am still using the iterator for the remove function. I can avoid using the iterator there too using the same principle, but it would mean searching for an object twice, once to read and modify it and once to erase it if necessary. I think that keeping the iterator for the remove function is worthwhile, but I'm not sure. Am I wrong in this?

Reply:
I am agree